zipline/data/us_equity_pricing.py

```diff
@@ -528,6 +528,16 @@ def _spot_col(self, colname):
     def get_last_traded_dt(self, asset, day):
         volumes = self._spot_col('volume')
         search_day = day
+
+        if search_day >= asset.end_date:
+            # move to the last day before the asset end date
+            try:
+                search_day = self._calendar[
+                    self._calendar.get_loc(asset.end_date) - 1
```

Comment:
I rewrote that chunk of code, agreed with @ssanderson that the try/except was too broad.  Now, we simply go back to the last trading day before the asset end date, if needed.

```
        if day >= asset.end_date:
            # go back to one day before the asset ended
            search_day = self._calendar[
                self._calendar.get_loc(asset.end_date) - 1
            ]
        else:
            search_day = day
```